### PR TITLE
Boxplot for Framework V/s Metrics

### PR DESCRIPTION
**Pull request by shreyash2106**
This PR addresses the need for a Boxplot of Frameworks V/s various metrics such as loss, training/inference time. This is similar to the plot shown in the AMLB paper: https://arxiv.org/pdf/2207.12560.pdf - Pg. 21, Fig. 3

Below are screenshots depicting this change 
Screenshot 2023-06-30 at 11 15 39 AM
Screenshot 2023-06-30 at 11 15 45 AM

### CODE REVIEW

File: src/autogluon_dashboard/plotting/framework_boxplot.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+from .all_plots import Plot
+
+
+class FrameworkBoxPlot(Plot):
+    def __init__(
+        self,
+        plot_title: str,
+        dataset_to_plot: hvplot.Interactive,
+        plot_type: str = "hvplot",
+        x_axis: Union[str, list] = None,
```
Comment (yinweisu) [inline src/autogluon_dashboard/plotting/framework_boxplot.py:14]: I might have missed this in your previous PRs, but would recommend do List[type of elements] instead

File: src/autogluon_dashboard/plotting/framework_boxplot.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        y_axis: Union[str, list] = None,
+        graph_type: str = "box",
+        xlabel: str = "",
+        ylabel: str = "",
+        label_rot: int = 90,
+        table_cols: list = ...,
```
Comment (yinweisu) [inline src/autogluon_dashboard/plotting/framework_boxplot.py:20]: Is this a correct syntax by providing default list as `...`?

File: src/autogluon_dashboard/plotting/framework_boxplot.py (hunk 1 above)
Comment (shreyash2106) [reply src/autogluon_dashboard/plotting/framework_boxplot.py:14]: Sure!

File: src/autogluon_dashboard/plotting/framework_boxplot.py (hunk 2 above)
Comment (shreyash2106) [reply src/autogluon_dashboard/plotting/framework_boxplot.py:20]: I could do empty list to be more specific

**Review by yinweisu — APPROVED** [review]
LGTM

**Pull request merged**